### PR TITLE
Alerting: Fix alert form broken coming from panel (#64148)

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
Manual PR for backporting https://github.com/grafana/grafana/pull/64148

### CODE REVIEW

File: public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx [hunk 1]
```diff
@@ -153,18 +156,17 @@ export const AlertRuleForm: FC<Props> = ({ existing, prefill }) => {
...
-    const shouldReset = !existing && !prefill && defaultDsAndQueries.queries;
+    const shouldReset = !existing && !prefill && defaultDsAndQueries.queries && !Boolean(defaultsInQueryParams);
     if (shouldReset) {
       reset({
         ...getDefaultFormValues(),
         queries: defaultDsAndQueries.queries,
         condition: 'C',
         ...defaultsInQueryParamsObject,
         type: RuleFormType.grafana,
-        evaluateEvery: evaluateEvery,
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx:164]: Why was this line removed in this backport?

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx:164]: there was a bug fix (not sure why is not in this branch) removing the evaluateEvery in here and in the array of dependencies

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx:164]: I got a merge conflict on the array of dependencies

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx:164]: https://github.com/grafana/grafana/pull/63763

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx:164]: aha!, the backport was not merged yet https://github.com/grafana/grafana/pull/63794 @gillesdemey

File: public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ import {
   Input,
   LoadingPlaceholder,
   Spinner,
-  useStyles2,
+  useStyles2
```
[the remark this answers is not in the document]
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx]: actually it was complaining about the import order

**Review by gillesdemey — APPROVED** [review]
